tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,89 @@
+import dataclasses
+import io
+import os
+import zipfile
+from typing import List, Optional
+
+import click
+import junit_xml
+
+import tmt.steps.report
+import tmt.utils
+
+from . import junit
+
+
+@dataclasses.dataclass
+class ReportPortalData(tmt.steps.report.ReportStepData):
+    url: Optional[str] = None
+    token: Optional[str] = None
+    launch_name: Optional[str] = None
+
+
+@tmt.steps.provides_method("reportportal")
+class ReportPortal(tmt.steps.report.ReportPlugin):
+    """
+    Send results in JUnit format to Report Portal instance.
+    """
+
+    _data_class = ReportPortalData
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options """
+        return [
+            click.option(
+                "--url", envvar="TMT_REPORT_REPORTPORTAL_URL", metavar="URL", required=True,
+                help="The URL of Report Portal instance where the data should be sent to"),
+            click.option(
+                "--token", envvar="TMT_REPORT_REPORTPORTAL_TOKEN", metavar="TOKEN", required=True,
+                help="The token to use for upload to Report Portal instance"),
+            click.option(
+                "--launch-name", metavar="NAME",
+                help="The launch name"),
+            ] + super().options(how)
+
+    def go(self) -> None:
+        """
+        Read executed tests, prepare junit, compress it to a zip zile and
+        send it to Report Portal instance.
+        """
+
+        super().go()
+
+        suite = junit.make_junit_xml(self)
+        data = junit_xml.TestSuite.to_xml_string([suite])
+        bytestream = io.BytesIO()
+
+        with zipfile.ZipFile(bytestream, "w", compression=zipfile.ZIP_DEFLATED,
+                             compresslevel=1) as zipstream:
+            # XML file names are irrelevant to Report Portal
+            with zipstream.open("tests.xml", "w") as entry:
+                entry.write(data.encode("utf-8"))
+        bytestream.seek(0)
+
+        with tmt.utils.retry_session() as session:
+            assert self.step.plan.my_run is not None
+            assert self.step.plan.my_run._workdir is not None
+            launch_name = self.get("launch_name") or os.path.basename(
+                self.step.plan.my_run._workdir)
+            response = session.post(
+                self.get("url"),
+                headers={
+                    "Authorization": "bearer " + self.get("token"),
+                    "accept": "*/*",
+                    },
+                files={
+                    # The zip filename is used as the launch name in Report Portal
+                    "file": (launch_name + ".zip", bytestream, "application/zip"),
+                    },
+                )
+
+        if not response.ok:
+            message = response.json().get("message")
```

Comment:
Tried to report results and got the following traceback:

```
  File "/home/psss/git/tmt/tmt/base.py", line 1514, in go
    step.go()
  File "/home/psss/git/tmt/tmt/steps/report/__init__.py", line 102, in go
    cast(Union[Action, ReportPlugin], phase).go()
  File "/home/psss/git/tmt/tmt/steps/report/reportportal.py", line 83, in go
    message = response.json().get("message")
  File "/home/psss/.virtualenvs/tmt/lib/python3.10/site-packages/requests/models.py", line 910, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib64/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
```

The error message was this:

```
(Pdb) response.text
'<html>\r\n<head><title>405 Not Allowed</title></head>\r\n<body>\r\n<center><h1>405 Not Allowed</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n'
```

Reply:
I was able to make it working if added `/api/v1/baseosqe/launch/import` as the suffix to the `url` as suggested by @brakogh. @mkoncek, could you please add a `--project` option and corresponding environment variable which would be used to construct the proper `url`?